shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -44,6 +44,38 @@
 NSNotificationName const FlutterViewControllerShowHomeIndicator =
     @"FlutterViewControllerShowHomeIndicator";
 
+/**
+ * Compute the interpolated value under linear interpolation.
+ */
+CGFloat FLTLinearInterpolatedValue(double progress, CGFloat from, CGFloat to, CGFloat scale) {
+  NSCAssert(progress >= 0 && progress <= 1, @"progress must be between 0 and 1");
+  return (from * (1 - progress) + to * progress) * scale;
+}
+
+/**
+ * Interpolate the viewport metrics for smoother rotation transition.
+ */
+void FLTInterpolateViewportMetrics(flutter::ViewportMetrics& viewportMetrics,
```

Comment:
Instead of passing in a viewportMetrics reference, can this instead be a method on the view controller that directly accesses `_viewportMetrics`?  Or was that too difficult to test?

Reply:
Yes I had issue with testing - `OCMock/NSInvocation` only supports populating NSObject references, and not structs. And I do not want to expose `viewportMetrics` property as it's anti-pattern. So I made this pure function for easier testing.